Security/SessionDownloadStrategy.php

```diff
@@ -39,14 +43,31 @@ class SessionDownloadStrategy implements DownloadStrategyInterface
     protected $sessionKey = 'sonata/media/session/times';
 
     /**
-     * @param TranslatorInterface $translator
-     * @param ContainerInterface  $container
-     * @param int                 $times
+     * @var Session
      */
-    public function __construct(TranslatorInterface $translator, ContainerInterface $container, $times)
+    private $session;
+
+    /**
+     * @param TranslatorInterface        $translator
+     * @param ContainerInterface|Session $session
+     * @param int                        $times
+     */
+    public function __construct(TranslatorInterface $translator, $session, $times)
     {
+        // NEXT_MAJOR : remove this block and set session from parameter.
+        if ($session instanceof ContainerInterface) {
+            @trigger_error(
+                'Using an instance of Symfony\Component\DependencyInjection\ContainerInterface is deprecated since 
+                version 3.x and will be removed in 4.0. Use Symfony\Component\HttpFoundation\Session\Session instead.',
+                E_USER_DEPRECATED
+            );
+
+            $this->session = $session->get('session');
+        } else {
+            $this->session = $session;
```

Comment:
Shouldn't we check the type here? Since if it's not a ContainerInterface but instead a [insert random class name here], it will fail?

Something like: 

```
elseif ($session instanceof SessionInterface) {
    $this->session = $session
} else {
    throw new \InvalidArgumentException()
}
```